internal/kubernetes/drainer.go

```diff
@@ -381,6 +387,34 @@ func (d *APICordonDrainer) Uncordon(n *core.Node, mutators ...nodeMutatorFn) err
 	return nil
 }
 
+// MarkDrain set a condition on the node to mark that the drain is scheduled.
+func (d *APICordonDrainer) MarkDrainDelete(n *core.Node) error {
+	nodeName := n.Name
+	// Refresh the node object
+	freshNode, err := d.c.CoreV1().Nodes().Get(nodeName, meta.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+	newConditions := []core.NodeCondition{}
+	for _, condition := range freshNode.Status.Conditions {
+		if string(condition.Type) != ConditionDrainedScheduled {
```

Comment:
Looks like this loop is removing all conditions except ConditionDrainedScheduled, and I believe your intent is the opposite.
```suggestion
		if string(condition.Type) == ConditionDrainedScheduled {
```